### PR TITLE
Add stall change side for loader

### LINKED ISSUE #39: Loader should not use CommandScheduler.getInstance().timeSinceScheduled()
as it doesn't support CommandGroups

### PR DESCRIPTION
**Pull request by EliyaFishman**
Fixes #39

### CODE REVIEW

**Comment by EliyaFishman** [comment]
[Stall change side on Loader](https://trello.com/c/CO74aQEi/126-stall-change-side-on-loader)

File: src/main/java/frc/robot/subsystems/loader/SetLoaderVelocity.java [hunk 1]
```diff
@@ -1,27 +1,32 @@
...
 import edu.wpi.first.wpiutil.math.MathUtil;
+import frc.robot.utils.DriverStationLogger;
 import java.util.function.DoubleSupplier;
 
 import static frc.robot.Robot.loader;
 import static frc.robot.Robot.robotConstants;
 
-public class setLoaderVelocity extends CommandBase {
+public class SetLoaderVelocity extends CommandBase {
     private PIDController pidController;
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/loader/SetLoaderVelocity.java]: Change to trigon PID controller

**Review by HilelAmiel — CHANGES_REQUESTED** [review]
one change

File: src/main/java/frc/robot/subsystems/loader/SetLoaderVelocity.java [hunk 2]
```diff
@@ -62,15 +67,29 @@ public SetLoaderVelocity(DoubleSupplier desiredVelocity, boolean isTuning) {
     @Override
     public void initialize() {
         if (isTuning)
-            SmartDashboard.putData("Loader pid controller", pidController);
+            SmartDashboard.putData("PID/Loader PID controller", pidController);
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/loader/SetLoaderVelocity.java]: update from master use TrigonPidController

**Review by HilelAmiel — CHANGES_REQUESTED** [review]
do the one thing and then I will see the rest of the code

**Comment by EliyaFishman** [comment]
Fixes #39

File: src/main/java/frc/robot/subsystems/loader/SetLoaderVelocity.java [hunk 3]
```diff
@@ -29,26 +34,42 @@ public SetLoaderVelocity(double desiredVelocity) {
         feedforward = robotConstants.controlConstants.loaderFeedforward;
         pidController = new TrigonPIDController(robotConstants.controlConstants.loaderPidSettings, desiredVelocity);
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/loader/SetLoaderVelocity.java:35]: pidController = new TrigonPIDController(robotConstants.controlConstants.loaderPidSettings, desiredVelocity);
if we dont want to calibrate? and make desiredVelocity suppliar?

File: src/main/java/frc/robot/subsystems/loader/SetLoaderVelocity.java [hunk 4]
```diff
@@ -29,26 +34,42 @@ public SetLoaderVelocity(double desiredVelocity) {
...
-        loader.setVoltage(
-            pidController.calculate(loader.getVelocity(), -6, 6) + feedforward.calculate(pidController.getSetpoint()));
+        if (Timer.getFPGATimestamp() - firstStallTimestamp < robotConstants.loaderConstants.kSpinBackwardsTime) {
+            loader.move(-feedforward.calculate(pidController.getSetpoint()));
+            inStall = false;
+            return;
+        }
+        if (!inStall && loader.getIsInStall() && Timer.getFPGATimestamp() - initializeTimestamp >
+            robotConstants.loaderConstants.kTimeout) {
+            DriverStationLogger.logToDS("A cell got stuck in the loader. Trying to unjam it now");
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/loader/SetLoaderVelocity.java]: DriverStationLogger.logToDS("A cell got stuck in the loader. Trying to unjam it now");
what is **unjam**? and remove **now**

File: src/main/java/frc/robot/subsystems/loader/Loader.java [hunk 5]
```diff
@@ -39,4 +41,8 @@ public void setVoltage(double voltage) {
     public void move(double power) {
         talonSRX.set(power);
     }
+
+  public boolean getIsInStall() {
+    return talonSRX.getStatorCurrent() > robotConstants.loaderConstants.kThresholdLimit;
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/loader/Loader.java]: kThresholdLimit is problematic to use... make another constant or calculate the stall by given voltage.
**V = I × R**

**Review by HilelAmiel — CHANGES_REQUESTED** [review]
few things

File: src/main/java/frc/robot/subsystems/loader/SetLoaderVelocity.java (hunk 4 above)
Comment (EliyaFishman) [reply src/main/java/frc/robot/subsystems/loader/SetLoaderVelocity.java]: [Here is defenition](https://www.yourdictionary.com/unjam)
It basically means that it's making it not stuck

File: src/main/java/frc/robot/subsystems/loader/SetLoaderVelocity.java (hunk 3 above)
Comment (EliyaFishman) [reply src/main/java/frc/robot/subsystems/loader/SetLoaderVelocity.java:35]: for calibrating we have the constructor:
``` java
    /**
     * This class accelerates the loader subsystem to the desired velocity using PID.
     * This constructor is used for tuning PID using the dashboard
     */
    public SetLoaderVelocity() {
        addRequirements(loader);
        feedforward = robotConstants.controlConstants.loaderFeedforward;
        pidController = new TrigonPIDController("Loader PID controller");
    }
```

Comment (EliyaFishman) [reply src/main/java/frc/robot/subsystems/loader/SetLoaderVelocity.java:35]: Why would we want to make desiredVelocity a supplier? It shouldn't change after we calibrated it

File: src/main/java/frc/robot/subsystems/loader/Loader.java [hunk 6]
```diff
@@ -17,12 +18,13 @@
...
     public Loader() {
         talonSRX = new WPI_TalonSRX(robotConstants.can.LOADER_TALON_SRX);
-        talonSRX.configClosedloopRamp(robotConstants.loaderConstants.kRampRate);
+        talonSRX.configOpenloopRamp(robotConstants.loaderConstants.kRampRate);
         talonSRX.setNeutralMode(NeutralMode.Coast);
         talonSRX.configSupplyCurrentLimit(
             new SupplyCurrentLimitConfiguration(true, robotConstants.loaderConstants.kCurrentLimit,
                 robotConstants.loaderConstants.kThresholdLimit, robotConstants.loaderConstants.kTimeout));
-    }
+        talonSRX.configSelectedFeedbackSensor(TalonSRXFeedbackDevice.CTRE_MagEncoder_Relative, 0, 0);
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/loader/Loader.java]: do an update from master

**Review by HilelAmiel — CHANGES_REQUESTED** [review]
~update time~

**Review by HilelAmiel — APPROVED** [review]

**Pull request merged**